Use callbacks in Neural Network Classifier and Regressor tutorial (#195)

epelaaez opened the pull request:
<!--
⚠️ If you do not respect this template, your pull request will be closed.
⚠️ Your pull request title should be short detailed and understandable for all.
⚠️ Also, please add it in the CHANGELOG file under Unreleased section.
⚠️ If your pull request fixes an open issue, please link to the issue.

✅ I have added the tests to cover my changes.
✅ I have updated the documentation accordingly.
✅ I have read the CONTRIBUTING document.
-->

### Summary
Integrates callbacks into the Neural Network Classifier and Regressor tutorial to make it more interactive and demonstrate the use of the newly added `callback` parameter.

### Details and comments
I added an example of the live plot I mentioned in the linked issue (closes #195). Let me know any feedback and/or things that need to be changed so I can go ahead and implement it for the four classes present in this tutorial.

adekusar-drl commented on docs/tutorials/02_neural_network_classifier_and_regressor.ipynb:
```diff
@@ -152,51 +154,80 @@
    "metadata": {},
    "outputs": [],
    "source": [
+    "# callback function that draws a live plot when the .fit() method is called\n",
+    "def callback(x, y):\n",
+    "    clear_output(wait=True)\n",
+    "    objective_func_vals.append(y)\n",
+    "    plt.title(\"Objective function value against iteration for NeuralNetworkClassifier\")\n",
+    "    plt.xlabel(\"Iteration\")\n",
+    "    plt.ylabel(\"Objective function value\")\n",
+    "    plt.plot(range(len(objective_func_vals)), objective_func_vals)\n",
+    "    plt.show()\n",
+    "\n",
```
I'd rather add a text cell explaining what callback is and how it can be used. Other than that looks nice. I hope, the callback function can be reused across different models.

epelaaez replied:
I've added a text cell explaining the callback function just before a code cell only containing the callback function. Let me know if the content in the text cell is alright or if it should be changed.

adekusar-drl reviewed (commented):
Thanks, looks great! This is what was expected, if you could add plots for other models that would be perfect.

epelaaez commented:
I added the live plot for all the models in the tutorial.

I've also added a text cell talking about the callback function just before the first model. Let me know if the content in this text cell is okay or if I should make any changes. Furthermore, the same callback function is passed to all the models to avoid declaring it before every model.

adekusar-drl reviewed (approved):
Thanks, looks good to me!
